src/main/java/Controller.java

```diff
@@ -2610,6 +2614,60 @@ void saveFlags() {
         Log.LogMessage("Flags Saved");
     }
 
+    @FXML
+    void saveThisFlag(Title title)
+    {
+        ZonedDateTime startOfToday = LocalDate.now().atStartOfDay(ZoneId.systemDefault());
+        long todayMillis = startOfToday.toEpochSecond() * 1000;
+        Date today = new Date(todayMillis);
+        PreparedStatement s = null;
+        String sql = """
+            UPDATE Titles
+            SET FLAGGED = TRUE, DATE_FLAGGED = ?, ISSUE_FLAGGED = ?
+            WHERE TITLEID = ?
+            """;
+        try {
+            s = conn.prepareStatement(sql);
+            s.setString(1, DateFormat.getDateInstance().format(today));
+            if (title.getIssueFlagged() == 0) {
+                s.setString(2, null);
+            } else {
+                s.setString(2, Integer.toString(title.getIssueFlagged()));
+            }
+            s.setString(3, Integer.toString(title.getId()));
+            s.executeUpdate();
+            s.close();
+        } catch (SQLException sqlExcept) {
+            sqlExcept.printStackTrace();
+        }
+        invalidateOrders();
```

Comment:
invalidateTitles() would make more sense than invalidateOrders() here

Reply:
I will add to bug report list in Section 4! Thank you!

Reply:
> When exiting, if you do not click save flags then it asks you to confirm you still wish to leave
> Flags have not been saved warning when trying to add/edit/delete title

Removed the boolean unsaved check from the Main function, and the add/edit/delete title functions

> Clicking save flags button will remove some items that were saved -> Not entirely sure why

Unable to duplicate

> Flagging title does not update the last flagged date or remove the “has not been flagged in 6 months” warning - If you click away, it will update the flagged date but not the warning
If you flag a title in a search, when you get rid of the search, it sometimes does not show as flagged even if it is in the db and if you search again it does not show as flagged yet

Should be fixed through editing invalidateOrders into invalidateTitles